cache/sampleStoreInit.js

```diff
@@ -41,15 +64,20 @@ function getPreviousStatus() {
 } // persistInProgress
 
 /**
- * Clear all the "sampleStore" keys (for subjects, aspects, samples) from
- * redis.
+ * Clear all the "sampleStore" keys (for subjects, aspects, samples, subaspmap)
+ * from redis.
  *
  * @returns {Promise} upon completion.
  */
 function eradicate() {
   const promises = Object.getOwnPropertyNames(constants.indexKey)
     .map((s) => redisClient.smembersAsync(constants.indexKey[s])
     .then((keys) => {
+      if (constants.indexKey[s] === constants.indexKey.sample) {
+        // this is done to delete keys prefixed with "samsto:subaspmap:"
+        keys.push(...getSubAspMapKeys(keys));
+      }
```

Comment:
Yeah, maybe add another master list of these keys. But this is good for now, to not block enabling the feature.